annotated_types/__init__.py

```diff
@@ -296,6 +297,19 @@ class Timezone(BaseMetadata):
     tz: Union[str, tzinfo, EllipsisType, None]
 
 
+@dataclass(frozen=True, **SLOTS)
+class Quantity(BaseMetadata):
+    """Indicates that the value is a physical quantity with the specified unit.
+
+    It is intended for usage with numeric types (any instance of ``numbers.Number``),
```

Comment:
I would not mention the `numbers` module in a type checking context, there are too many problems. I think for this docstring it's fine to just say "numeric type" without precision; it's up to consumers of the library to decide exactly what to expect.